core/src/main/java/org/vertexium/query/CompositeGraphQuery.java

```diff
@@ -106,6 +107,56 @@ protected boolean isIncluded(Element element) {
     }
 
     @Override
+    public QueryResultsIterable<ExtendedDataRow> extendedDataRows() {
+        return extendedDataRows(FetchHint.ALL);
+    }
+
+    @Override
+    public QueryResultsIterable<ExtendedDataRow> extendedDataRows(EnumSet<FetchHint> fetchHints) {
+        final Set<ExtendedDataRowId> seenIds = new HashSet<>();
+        return new QueryResultsSelectManyIterable<ExtendedDataRow>(this.queries) {
+            @Override
+            public Iterable<ExtendedDataRow> getIterable(Query query) {
+                return query.extendedDataRows(fetchHints);
+            }
+
+            @Override
+            protected boolean isIncluded(ExtendedDataRow extendedDataRows) {
+                if (seenIds.contains(extendedDataRows.getId())) {
+                    return false;
+                }
+                seenIds.add(extendedDataRows.getId());
+                return super.isIncluded(extendedDataRows);
+            }
+        };
+    }
+
+    @Override
+    public QueryResultsIterable<? extends VertexiumObject> search(EnumSet<VertexiumObjectType> objectTypes, EnumSet<FetchHint> fetchHints) {
+        final Set<Object> seenIds = new HashSet<>();
+        return new QueryResultsSelectManyIterable<VertexiumObject>(this.queries) {
+            @Override
+            public Iterable<? extends VertexiumObject> getIterable(Query query) {
+                return query.search(objectTypes, fetchHints);
+            }
+
+            @Override
```

Comment:
It is not very clear why a previously "seen" Id, should trigger the isIncluded flag to be false. Add comment or make SeenIds variable name a bit more descriptive.